### PR TITLE
Use pytest instead of unittest

### PR DESCRIPTION
**Pull request by Tapot**
Use Pytest instead of unittest python module
- add pytest==6.2.5 to dev requirements
- add pytest settings to setup.cfg - just specify main python code place
- update CONTRIBUTING.md about tests run + some description typo fixes in the section
- update github actions for running tests
- update tests syntax from uniitests to more-pytest style - use asserts instead of self.assert methods

### CODE REVIEW

**Comment by burkovae** [comment]
If there is a decision to move to `pytest`, please consider moving tests into a separate directory as well. (As for the test directory structure, there're strategies as well)

**Comment by Tapot** [comment]
> If there is a decision to move to `pytest`, please consider moving tests into a separate directory as well. (As for the test directory structure, there're strategies as well)

I think that it is a better way too, it can be the second step in s separate PR - for better diff readability

Let's wait for a maintainer opinion

File: evidently/analyzers/test_utils.py [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-import unittest
 
 import pandas
 
 from evidently.analyzers.utils import process_columns
 from evidently.pipeline.column_mapping import ColumnMapping
 
 
-class TestUtils(unittest.TestCase):
+class TestUtils:
```
Comment (Liraim) [inline evidently/analyzers/test_utils.py:9]: class can be removed, and all methods can be made just regular top level functions.

File: evidently/utils/test_utils.py [hunk 2]
```diff
@@ -1,38 +1,40 @@
...
-class TestNumpyEncoder(unittest.TestCase):
+@pytest.fixture
+def numpy_encoder():
+    return NumpyEncoder()
 
-    def setUp(self) -> None:
-        self.encoder = NumpyEncoder()
 
-    def test_int_convert(self):
+class TestNumpyEncoder:
```
Comment (Liraim) [inline evidently/utils/test_utils.py:13]: same here, class can be removed.

**Review by Liraim — CHANGES_REQUESTED** [review]

File: evidently/analyzers/test_utils.py (hunk 1 above)
Comment (Tapot) [reply evidently/analyzers/test_utils.py]: I've moved the tests to functions, classes are removed for already moved to pyrest test

**Comment by Tapot** [comment]
I'm going to move new tests to pytest in the PR and I can move all tests to one folder. Or maybe moving tests to a separate folder will be better in another PR, @Liraim what do you think?

**Comment by Liraim** [comment]
> I'm going to move new tests to pytest in the PR and I can move all tests to one folder. Or maybe moving tests to a separate folder will be better in another PR, @Liraim what do you think?

I will move tests to separate folder in another PR. So nothing is required here.

**Review by Liraim — APPROVED** [review]
Looks good.

**Review by Liraim — CHANGES_REQUESTED** [review]
Need fix for linter.

**Review by Liraim — APPROVED** [review]
Now Looks good.

**Pull request merged**